### PR TITLE
Optional branchID in gsim logic tree files

### PR DESCRIPTION
**Pull request by micheles**
We discussed this a long time ago. I am implementing it now since I am annoyed by non-unique-branch-id warnings in 5 models of the mosaic: AUS, CCA, HAW, SAM, SEA. Next step would be to remove the branch IDs from such models and have them to be autogenerated,

### CODE REVIEW

File: openquake/qa_tests_data/scenario/case_13/expected/branches.org [hunk 1]
```diff
@@ -0,0 +1,19 @@
++-----------+--------+----------------------------+
+| branch_id | abbrev | uvalue                     |
```
Comment (mmpagani) [inline openquake/qa_tests_data/scenario/case_13/expected/branches.org:2]: Are you explaining somewhere the difference between abbrev and branch_id?

**Review by mmpagani — COMMENTED** [review]
I think it's interesting. What I do not understand is how you plan to handle cases where we use 'apply to branch'

**Comment by micheles** [comment]
`applyToBranch` is not supported for the gsim logic tree file and I am only considering that ;-)

**Comment by mmpagani** [comment]
> `applyToBranch` is not supported for the gsim logic tree file and I am only considering that ;-)

Oh sorry I should have read more carefully the PR title

**Comment by mmpagani** [comment]
B

> > `applyToBranch` is not supported for the gsim logic tree file and I am only considering that ;-)
> 
> Oh sorry I should have read more carefully the PR title

However, although it's very rare there could be the possibility of using apply to branch also in the GMM LT

**Comment by micheles** [comment]
> However, although it's very rare there could be the possibility of using apply to branch also in the GMM LT

If needed, we will implement it, and then the engine will simply force people to specify the branch IDs explicitly if applyToBranch is used.

**Pull request merged**